src/main/scala/dev/kamu/cli/commands/PullCommand.scala

```diff
@@ -20,15 +20,44 @@ class PullCommand(
   private val logger = LogManager.getLogger(getClass.getName)
 
   def run(): Unit = {
-    val datasets =
-      if (all)
-        metadataRepository.getAllDatasets()
-      else
-        datasetIDs
-          .map(DatasetID)
-          .map(metadataRepository.getDataset)
-
-    val numUpdated = datasets
+    val s = metadataRepository.getAllDatasetIDs()
+    val datasets: IndexedSeq[DatasetID] = s.toIndexedSeq
+    val datasetIntID = s.zipWithIndex.toMap
+    val idToInt: DatasetID => Int = d => datasetIntID(d)
+    val intToId: Int => DatasetID = d => datasets(d)
+    val idToDataset: DatasetID => Dataset = d => metadataRepository.getDataset(d)
+
+    val scheduled: Array[Boolean] = new Array(datasets.length)
+    for (i <- scheduled.indices) scheduled.update(i, false)
+    var toSchedule: List[Int] = Nil
+    var plan: Seq[Int] = Nil
+    def isScheduled(d: Int): Boolean = scheduled(d)
```

Comment:
better not to mix different styles of function declaration in one function, pick either `val x = ...` or `def x() = ...`